scapy/layers/bluetooth.py

```diff
@@ -477,7 +580,10 @@ class HCI_LE_Meta_Advertising_Report(Packet):
                     ByteEnumField("atype", 0, {0:"public", 1:"random"}),
                     LEMACField("addr", None),
                     FieldLenField("len", None, length_of="data", fmt="B"),
-                    StrLenField("data", "", length_from=lambda pkt:pkt.len), ]
+                    PacketListField("data", [], EIR_Hdr,
+                                    length_from=lambda pkt:pkt.len),
+                    SignedCharField("rssi", 0)]
```

Comment:
BTW, something else that's incorrect here, is that the each of these fields are supposed to be arrays. The Number field indicates the length of the arrays. I don't know of any Bluetooth chip that actually uses this feature though – all of the chips I've seen set Number to 1.

I didn't try to fix this because I'm not sure what Field types to use for that.